### PR TITLE
Typos: `test/SILGen`

### PR DESCRIPTION
**Pull request by Sajjon**
Fix typos in `test/SILGen`

This is one batch of many PRs fixing typos, [see the tracking issue.](https://github.com/swiftlang/swift/issues/75022)

### CODE REVIEW

**Comment by AnthonyLatsis** [comment]
Conflict creeped in.

File: test/SILGen/dynamically_replaceable.swift [hunk 1]
```diff
@@ -9,22 +9,22 @@ defer {
 }
 
 // CHECK-LABEL: sil hidden [ossa] @$s23dynamically_replaceable014maybe_dynamic_B0yyF : $@convention(thin) () -> () {
-// IMPLICIT-LABEL: sil hidden [dynamically_replacable] [ossa] @$s23dynamically_replaceable014maybe_dynamic_B0yyF : $@convention(thin) () -> () {
+// IMPLICIT-LABEL: sil hidden [dynamically_replaceable] [ossa] @$s23dynamically_replaceable014maybe_dynamic_B0yyF : $@convention(thin) () -> () {
```
Comment (Sajjon) [inline test/SILGen/dynamically_replaceable.swift:12]: @AnthonyLatsis I guess I should revert ALL changes to `dynamically_replaceable` in this PR as well?

File: test/SILGen/package_allow_non_resilient_access.swift [hunk 2]
```diff
@@ -44,7 +44,7 @@
 // RUN:   -experimental-allow-non-resilient-access \
 // RUN:   -emit-module -emit-module-path %t/Utils.swiftmodule \
 // RUN: 2>&1 | %FileCheck %s --check-prefix=CHECK-DIAG-1
-// CHECK-DIAG-1: warning: ignoring -experimental-skip-non-exportable-decls (overriden by -allow-non-resilient-access)
+// CHECK-DIAG-1: warning: ignoring -experimental-skip-non-exportable-decls (overridden by -allow-non-resilient-access)
```
Comment (AnthonyLatsis) [inline test/SILGen/package_allow_non_resilient_access.swift:47]: This is a FileCheck directive that checks for an occurrence of the given string, not just a comment, so you need to update the warning message too (`warn_ignore_option_overriden_by`). 


Ditto other changes in this file.

File: test/SILGen/dynamically_replaceable.swift (hunk 1 above)
Comment (AnthonyLatsis) [reply test/SILGen/dynamically_replaceable.swift:12]: Yeah, I suggest renaming the attribute in a separate patch.

Comment (Sajjon) [reply test/SILGen/dynamically_replaceable.swift]: reverted

File: test/SILGen/package_allow_non_resilient_access.swift (hunk 2 above)
Comment (Sajjon) [reply test/SILGen/package_allow_non_resilient_access.swift]: I reverted changes related to that in this file. I did not quite understand what you suggested I change, and I saw references to `warn_ignore_option_overriden_by` in some other files, so felt best revert:

Screenshot 2024-08-08 at 19 25 39

**Comment by Sajjon** [comment]
@AnthonyLatsis I think this is merge ready?

File: test/SILGen/initializers.swift [hunk 3]
```diff
@@ -800,7 +800,7 @@ class ThrowDerivedClass : ThrowBaseClass {
   // CHECK:   try_apply [[UNWRAP_FN]]({{%.*}}) : $@convention(thin) (Int) -> (Int, @error any Error), normal [[NORMAL_BB:bb[0-9]+]], error [[ERROR_BB:bb[0-9]+]]
   //
   // Now we emit the call to the initializer. Notice how we return self back to
-  // its memory locatio nbefore any other work is done.
+  // its memory locations nbefore any other work is done.
```
Comment (adrian-prantl) [inline test/SILGen/initializers.swift:803]: It really looks like this is meant to be

```suggestion
  // its memory location before any other work is done.
```

no?

**Review by adrian-prantl — CHANGES_REQUESTED** [review]

File: test/SILGen/initializers.swift (hunk 3 above)
Comment (Sajjon) [reply test/SILGen/initializers.swift]: @adrian-prantl thx, right you are. Fixed!

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Review by AnthonyLatsis — APPROVED** [review]

**Review by adrian-prantl — APPROVED** [review]

**Pull request merged**